### PR TITLE
Admin Update hotel

### PR DESCRIPTION
**Pull request by karify98**
- Add Function Update hotel
- Fix Show hotel and test show hotel

### CODE REVIEW

File: app/Http/Controllers/Admin/HotelController.php [hunk 1]
```diff
@@ -144,4 +146,86 @@ public function destroy($id)
...
+            'address',
+            'star',
+            'introduce',
+            'place_id'
+        ];
+
+        $with['place'] = function ($query) {
+            $query->select('id', 'name');
+        };
+        $with['rooms'] = function ($query) {
```
Comment (ghost) [inline app/Http/Controllers/Admin/HotelController.php]: room ko dùng thì bỏ đi cũng dc em.
cái nào sử dụng thì mới query ra thôi

File: app/Http/Controllers/Admin/HotelController.php [hunk 2]
```diff
@@ -144,4 +146,86 @@ public function destroy($id)
...
+            $query->select('hotel_id', 'id', 'name');
+        };
+        $with['images'] = function ($query) {
+            $query->select();
+        };
+        $with['hotelServices'] = function ($query) {
+            $query->select('id', 'hotel_id', 'service_id');
+        };
+        $with['hotelServices.service'] = function ($query) {
```
Comment (ghost) [inline app/Http/Controllers/Admin/HotelController.php]: cái này cũng ko dùng nên bỏ ra luôn.

**Comment by karify98** [comment]
e mới sửa lại với update thêm phần giao diện của page cho đẹp
a @at-huydq2  rv lại giúp em

File: app/Http/Controllers/Admin/HotelController.php [hunk 3]
```diff
@@ -147,4 +148,82 @@ public function destroy($id)
...
+        }
+        //save hotel services
+        $hotel->hotelServices()->saveMany($hotelServices);
+
+        if (isset($request->images)) {
+            Image::storeImages($request->images, 'hotel', $hotel->id, config('image.hotels.path_upload'));
+        }
+
+        if ($result) {
+            return redirect()->route('hotel.show', $id);
```
Comment (ghost) [inline app/Http/Controllers/Admin/HotelController.php]: I think we should have a message to notify that update successfully.

File: resources/views/backend/hotels/edit.blade.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                      <span class="text-danger">{{ $errors->first('place_id') }}</span>
+                    @endif
+                  </div>
+                   {{-- select star --}}
+                  <div class="form-group ml-5per" {{ $errors->has('star') ? ' has-error' : '' }}>
+                    <p class="font-weight-bold">{{ __('Star') }}</p>
+                    <select class="form-control" name="star">
+                      <option value="">{{ __('Star') }}</option>
+                      @for($i = App\Model\Hotel::STAR_MIN; $i <= App\Model\Hotel::STAR_MAX; $i++ )
+                        <option value="{{ $i }}" {{$hotel->star == $i ? 'selected' :''}}>{{ $i }}</option>
```
Comment (ghost) [inline resources/views/backend/hotels/edit.blade.php]: Space after `:` like this: `'selected' : ''`

File: resources/views/backend/hotels/edit.blade.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                </div>
+                {{-- select place adn star --}}
+                <div class="form-inline">
+                  {{-- place select --}}
+                  <div class="form-group" {{ $errors->has('place') ? ' has-error' : '' }}>
+                    <p class="font-weight-bold">{{ __('Place') }}</p>
+                    <select class="form-control" name="place_id">
+                      <option value="">{{ __('Choose Place') }}</option>
+                      @foreach($places as $place)
+                        <option value="{{ $place->id }}" {{$hotel->place_id == $place->id ? 'selected' :''}}>{{ $place->name }}</option>
```
Comment (ghost) [inline resources/views/backend/hotels/edit.blade.php]: Two thing here:
- One space after `:`.
- Should unify only one kind when put varriable in `{{ ... }}`.
  1. Wrapping by space: `{{ $varA }}`
  2. No wrapping by space: `{{$varB}}`
  I think we should use `#1` standard.

File: public/css/main.css [hunk 6]
```diff
@@ -280,3 +280,6 @@
     max-height:200px;
     display: block; 
 }
+.font-weight-bold {
```
Comment (ghost) [inline public/css/main.css]: không cần css cho cái này, dùng thẻ `<strong></strong>` được rồi

**Review by ghost — COMMENTED** [review]
Plz check my comment. Thanks.

File: resources/views/backend/hotels/edit.blade.php [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+                </div>
+                {{-- select place adn star --}}
+                <div class="form-inline">
+                  {{-- place select --}}
+                  <div class="form-group" {{ $errors->has('place') ? ' has-error' : '' }}>
+                    <p><strong>{{ __('Place') }}</strong></p>
+                    <select class="form-control" name="place_id">
+                      <option value="">{{ __('Choose Place') }}</option>
+                      @foreach($places as $place)
+                        <option value="{{ $place->id }}" {{$hotel->place_id == $place->id ? 'selected' :''}}>{{ $place->name }}</option>
```
Comment (ghost) [inline resources/views/backend/hotels/edit.blade.php]: Two thing here:
- One space after `:` like this: `'selected' : ''`.
- Should unify only one kind when put varriable in `{{ ... }}`.
  1. Wrapping by space: `{{ $varA }}`
  2. No wrapping by space: `{{$varB}}`
  I think we should use `#1` standard: `{{ $varA }}`.

File: resources/views/backend/hotels/edit.blade.php [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+                      <span class="text-danger">{{ $errors->first('place_id') }}</span>
+                    @endif
+                  </div>
+                   {{-- select star --}}
+                  <div class="form-group ml-5per" {{ $errors->has('star') ? ' has-error' : '' }}>
+                    <p><strong>{{ __('Star') }}</strong></p>
+                    <select class="form-control" name="star">
+                      <option value="">{{ __('Star') }}</option>
+                      @for($i = App\Model\Hotel::STAR_MIN; $i <= App\Model\Hotel::STAR_MAX; $i++ )
+                        <option value="{{ $i }}" {{ $hotel->star == $i ? 'selected' :'' }}>{{ $i }}</option>
```
Comment (ghost) [inline resources/views/backend/hotels/edit.blade.php]: Space after `:`

File: resources/views/backend/hotels/edit.blade.php [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+                  <label for="old-images">{{ __('Old Images') }}</label>
+                  <div
+                    id="old-images"
+                    class="col-md-12"
+                    data-token="{{ csrf_token() }}"
+                    data-title="{{ __('Confirm deletion!') }}"
+                    data-confirm="{{ __('Are you sure you want to delete?') }}">
+                    @if (isset($hotel->images[0]))
+                      @foreach ($hotel->images as $img)
+                        <div id="old-img-{{$img->id}}" class="col-md-3 text-center img-contain">
```
Comment (ghost) [inline resources/views/backend/hotels/edit.blade.php]: Unify space in `{{ ... }}`

**Comment by at-huydq2** [comment]
LGTM.

File: resources/views/backend/hotels/show.blade.php [hunk 10]
```diff
@@ -69,7 +72,7 @@
           @endforeach
         </ul>
         <div class="clearfix"></div>
-        <h3>{{ __('Rooms') . ' (' . $hotel->rooms->count() . ')' }}</h3>
+        <h3><a href="{{ route('room.index', $hotel->id) }}">{{ __('Rooms') . ' (' . $hotel->rooms->count() . ')' }}</a></h3>
```
Comment (ghost) [inline resources/views/backend/hotels/show.blade.php]: I think we should show text like this: `Total rooms: 8.`
And please be careful between `$hotel->rooms->count()` and `$hotel->rooms()->count()`.
1. `$post->comments->count();` loads the collection and returns count of its elements – redundant if all you need is the count.
2. `$post->comments()->count();` better, no collection loading, but runs the query everytime we use it.

**Review by ghost — COMMENTED** [review]
Please check my comment again. ;) :D

File: app/Http/Controllers/Admin/HotelController.php [hunk 11]
```diff
@@ -147,4 +147,87 @@ public function destroy($id)
...
+            }
+        }
+        //save hotel services
+        $saveServices = $hotel->hotelServices()->saveMany($hotelServices);
+
+        if (isset($request->images)) {
+            Image::storeImages($request->images, 'hotel', $hotel->id, config('image.hotels.path_upload'));
+        }
+
+        if ($result && $deleteOldServices && $saveServices) {
```
Comment (ghost) [inline app/Http/Controllers/Admin/HotelController.php]: dùng try catch đi e.

File: resources/views/backend/hotels/show.blade.php [hunk 12]
```diff
@@ -69,14 +74,7 @@
...
         <div class="clearfix"></div>
-        <h3>{{ __('Rooms') . ' (' . $hotel->rooms->count() . ')' }}</h3>
-        <ul>
-          @foreach ($hotel->rooms as $room)
-            <div class="col-md-3 pull-left">
-              <li><a href="">{{ $room->name }}</a></li>
-            </div>
-          @endforeach
-        </ul>
+        <h3><a href="{{ route('room.index', $hotel->id) }}">{{ __('Total rooms') . ': ' . $totalRooms . '.' }}</a></h3>
```
Comment (ghost) [inline resources/views/backend/hotels/show.blade.php]: Please change to this:
```php
__('Total rooms: :total.', ['total' => $totalRooms])
```

**Review by ghost — APPROVED** [review]
LGTM. Thanks.

**Comment by ghost** [comment]
LGTM

**Pull request merged**